_scripts/collect_contributor_data.R

```diff
@@ -44,11 +44,21 @@ repos <- lapply(repos, function(x) {
     # but keep an eye on https://github.com/ropenscilabs/allcontributors/issues/36 for a potentially
     # better solution.
     Sys.sleep(5)
-    return(one_repo[, c("logins", "avatar")])
+
+    # Add the repo name
+    one_repo$repo <- sprintf('%s/%s', org_name, x$name)
+
+    return(one_repo)
   }
 })
 
 ctbs <- do.call(rbind, repos)
+# Write out raw contributor data
+write.csv(ctbs, file = "_data/raw_epiverse_contributors.csv",
+          row.names = FALSE)
+
+# Retain only relevant info for about page
+ctbs <- ctbs[, c("logins", "avatar")]
 # Remove duplicates
 result <- ctbs[!duplicated(ctbs$logins), ]
```

Comment:
Thanks for the code! My initial idea is below:

```suggestion
ctbs |>
  dplyr::summarise(
    type = toString(type),
    repo = toString(repo),
    .by = logins
  )
```

I.e., still save a single file with one row per contributors but adding two extra cols with type of contributions. The main thing that we are interested in from a MEAL point of view is where people made their first contribution. I.e., the value of "type" and "repo" for new rows since the last time.

Does this make sense?

The good thing with your proposed approach is that it keeps the maximum of info, which we can always summarise later. My only potential concern is the number of diffs it'll generate. Can we maybe save the raw data in a google spreadsheet or similar?

Reply:
The **dplyr** operations make sense, although I am adding back avatar in the summarise (otherwise the pages don't have the information that is used at this time). This is done in 67f099ffc56a46bdfe32bbbc24e8bfbe77d9968e

> from a MEAL point of view is where people made their first contribution. I.e., the value of "type" and "repo" for new rows since the last time.

This makes sense – this indeed **only applies to new rows** as the order of events+repo's is arbitrary for existing contributors.

---

> Can we maybe save the raw data in a google spreadsheet or similar?

For sure – I don't have a google account so if that's something we want, I can only scaffold an initial version but not test it 👍

Reply:
I have a google account so maybe the simplest option is for me to take it from here. Is this okay with you?

Reply:
Yes 💯 

Seems like I lost track of this item, apologies for not pinging you about this earlier.

Reply:
Since I'm struggling a bit with authentification issues and there is value in already implementing the changes proposed here, I'll open a follow up issue to address this.